service/route53/internal/customizations/custom_error_deser_test.go

```diff
@@ -0,0 +1,122 @@
+		  </Messages>
+		  <RequestId>b25f48e8-84fd-11e6-80d9-574e0c4664cb</RequestId>
+		</InvalidChangeBatch>`),
+			expectedError:     "InvalidChangeBatch: ChangeBatch errors occurred",
+			expectedRequestID: "b25f48e8-84fd-11e6-80d9-574e0c4664cb",
+		},
+		"standardRestXMLError": {
+			responseStatus: 500,
+			responseBody: []byte(`<?xml version="1.0"?>
+		<ErrorResponse xmlns="http://route53.amazonaws.com/doc/2016-09-07/">
+		  <Error>
+		    <Type>Sender</Type>
+		    <Code>MalformedXML</Code>
+		    <Message>1 validation error detected: Value null at 'route53#ChangeSet' failed to satisfy constraint: Member must not be null</Message>
+		  </Error>
+		  <RequestId>b25f48e8-84fd-11e6-80d9-574e0c4664cb</RequestId>
+		</ErrorResponse>
+		`),
+			expectedError:     "1 validation error detected:",
+			expectedRequestID: "b25f48e8-84fd-11e6-80d9-574e0c4664cb",
+		},
+		"Success response": {
+			responseStatus: 200,
+			responseBody: []byte(`<?xml version="1.0" encoding="UTF-8"?>
+		<ChangeResourceRecordSetsResponse>
+   			<ChangeInfo>
+      		<Comment>mockComment</Comment>
+      		<Id>mockID</Id>
+   		</ChangeInfo>
+		</ChangeResourceRecordSetsResponse>`),
+			expectedResponseID: "mockID",
+		},
+	}
+
+	for name, c := range cases {
+		server := httptest.NewServer(http.HandlerFunc(
+			func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(c.responseStatus)
+				w.Write(c.responseBody)
+			}))
+		defer server.Close()
+
+		t.Run(name, func(t *testing.T) {
+			svc := route53.NewFromConfig(aws.Config{
+				Region: "us-east-1",
+				EndpointResolver: aws.EndpointResolverFunc(func(service, region string) (aws.Endpoint, error) {
+					return aws.Endpoint{
+						URL:         server.URL,
+						SigningName: "route53",
+					}, nil
+				}),
+				Retryer: aws.NoOpRetryer{},
+			})
+			resp, err := svc.ChangeResourceRecordSets(context.Background(), &route53.ChangeResourceRecordSetsInput{
+				ChangeBatch: &types.ChangeBatch{
+					Changes: []*types.Change{},
+					Comment: aws.String("mock"),
+				},
+				HostedZoneId: aws.String("zone"),
+			})
+
+			if err == nil && len(c.expectedError) != 0 {
+				t.Fatalf("expected err, got none")
+			}
+
+			if len(c.expectedError) != 0 {
+				if e, a := c.expectedError, err.Error(); !strings.Contains(a, e) {
+					t.Fatalf("expected error to be %s, got %s", e, a)
+				}
+
+				var awsResponseError responseError
+				if !errors.As(err, &awsResponseError) {
+					t.Fatalf("expected error to be of type %T, was not", awsResponseError)
+				}
+
+				if e, a := c.expectedRequestID, awsResponseError.ServiceRequestID(); !strings.EqualFold(e, a) {
+					t.Fatalf("expected request id to be %s, got %s", e, a)
+				}
+			}
+
+			if len(c.expectedResponseID) != 0 {
+				if e, a := c.expectedResponseID, *resp.ChangeInfo.Id; !strings.EqualFold(e, a) {
+					t.Fatalf("expected response to have id %v, got %v", e, a)
+				}
+			}
+
+		})
+	}
+}
+
+type responseError interface {
+	ServiceRequestID() string
+}
```

Comment:
Since this is only used once in the test case it could be defined inline instead of a named type.